nixos/modules/security/misc.nix

```diff
@@ -98,6 +98,17 @@ with lib;
           enters the guest.  May incur significant performance cost.
       '';
     };
+
+    security.enableX32Abi = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable the x32 subarchitecture on x86_64 kernels.
```

Comment:
do you plan to add a gnux32 target support to build packages in nixpkgs? Otherwise it might not be very useful. I can have a look at adding it as well.